### PR TITLE
Drop name from TokenizerFactory

### PR DESCRIPTION
**Pull request by nik9000**
Drops `TokenizerFactory#name`, replacing it with
`CustomAnalyzer#getTokenizerName` which is much better targeted at
its single use case inside the analysis API.

Drops a test that I would have had to refactor which is duplicated by
`AnalysisModuleTests`.

To keep this change from blowing up in size I've left two mostly
mechanical changes to be done in followups:
1. `TokenizerFactory` can now be entirely dropped and replaced with
`Supplier<Tokenizer>`.
2. `AbstractTokenizerFactory`'s ctor still takes a `String` parameter
where the name once was.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzer.java [hunk 1]
```diff
@@ -36,19 +37,27 @@
...
         this.charFilters = charFilters;
         this.tokenFilters = tokenFilters;
         this.positionIncrementGap = positionIncrementGap;
         this.offsetGap = offsetGap;
     }
 
+    /**
+     * The name of the tokenizer as configured by the user.
+     */
+    public String getTokenizerName() {
```
Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzer.java]: Why do we need this? The TokenizerFactory.name() was never used here before.

**Review by rjernst — APPROVED** [review]
LGTM, I just left one comment.

File: core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzer.java (hunk 1 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzer.java]: It was used in the analyze action.

**Comment by nik9000** [comment]
Thanks for the review @rjernst! I merged and added the followups to my list.

**Pull request merged**